src/olympia/blocklist/tests/test_mlbf.py

```diff
@@ -351,3 +353,30 @@ def test_generate_filter_not_raises_if_all_versions_blocked(self):
         self._blocked_addon(file_kw={'is_signed': False})
         assert mlbf.data.not_blocked_items == []
         mlbf.generate_and_write_filter()
+
+    def test_duplicate_guid_is_blocked(self):
+        """
+        Test that if there are addons with duplicated guids, and one is blocked,
+        the addon should be blocked and the other should not be included in not_blocked
+        """
+        version = '2.1'
+        reused_addon = addon_factory(
+            status=amo.STATUS_DELETED,
+            version_kw={'version': version},
+            file_kw={'is_signed': True},
+        )
+        addon, block = self._blocked_addon(
+            version_kw={'version': version},
+            file_kw={'is_signed': True},
+        )
+
+        reused_addon.update(guid=GUID_REUSE_FORMAT.format(addon.id))
+        reused_addon.addonguid.update(guid=addon.guid)
+
+        self._block_version(block, self._version(addon), soft=False)
+
+        mlbf = MLBF.generate_from_db('test')
+
+        (block_version,) = MLBF.hash_filter_inputs([(addon.guid, version)])
+        assert block_version in mlbf.data.blocked_items
+        assert block_version not in mlbf.data.not_blocked_items
```

Comment:
being paranoid: can we assert that `(addon.guid, version)` is in `fetch_all_versions_from_db(MLBF._version_excludes)` ?  i.e. would have been included otherwise?

Reply:
I'd rather not test the underlying function but the final data set. Or am I missing something in your ask?

Reply:
I was looking for a verification the test is set up correctly - that we have a case where there is a guid, version pair that is unblocked, and a pair that is blocked, but they don't both end up in final data set.  I suppose closer to a functional test than an integration test.